rita/src/rita_common/traffic_watcher/mod.rs

```diff
@@ -175,28 +211,60 @@ pub fn watch<T: Read + Write>(mut babel: Babel<T>, neighbors: &Vec<Neighbor>) ->
     }
 
     for ((ip, interface), bytes) in total_input_counters {
-        if destinations.contains_key(&ip)
-            && if_to_ip.contains_key(&interface)
-            && identities.contains_key(&if_to_ip[&interface])
-        {
-            let id = identities[&if_to_ip[&interface]].clone();
-            *debts.get_mut(&id).unwrap() -= (destinations[&ip].clone()) * bytes;
-        } else {
-            warn!("flow destination not found {}, {}", ip, bytes);
+        let state = (
+            identities.get(&ip),
+            destinations.get(&ip),
+            if_to_ip.get(&interface),
+        );
+        match state {
+            (Some(id), Some(_dest), Some(_ifip)) => match debts.get_mut(&id) {
+                Some(debt) => {
+                    *debt -= (destinations[&ip].clone()) * bytes;
+                }
+                // debts is generated from identities, this should be impossible
+                None => warn!("No debts entry for input entry id {:?}", id),
+            },
+            (Some(_id), Some(_), None) => warn!("This case should not be possible"),
```

Comment:
Should we group all the "This case should not be possible" cases at the end with a (_, _, _) ?

Reply:
I was thinking about that, but it made it hard for me to keep track of scenarios than this exhaustive binary countdown. don't really care much, I'll let others chime in and change it according to majority vote.